jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -228,21 +229,34 @@ def _on_display_units_changed(self, event={}):
                     else:
                         data = self.dataset.selected_dc_item
 
-                    with u.set_enabled_equivalencies(
-                            u.spectral() + u.spectral_density(self._cube_wave) +
-                            _eqv_flux_to_sb_pixel() +
-                            _eqv_pixar_sr(data.meta.get('_pixel_scale_factor', 1)
-                                          if data else 1)):
-                        self.background_value = bg.to_value(new_unit)
+                    if prev_unit != new_unit:
+
+                        # NOTE: I don't think all of these equivalencies are necessary,

```

Comment:
If this is kept as-is, please link follow-up ticket here. Thanks.

Reply:
In that case, should the note be updated or removed? Currently it sounds like someone needs to come back and refactor (again).

Reply:
the point still stands that i don't know if all those equivalencies will be needed, that part needs to be investigated still.

Reply:
I am confused. The old code should provide only the necessary equivalencies. But here, you changed it to use your function for consistency, which I understand. But then why do we still need to investigate? Either we use the old code or the new one here. What do you have in mind?

Reply:
my new function takes in a list of equivalencies. it also handles cases like converting between units in surface brightness per pixel squared that also need spectral density. that conversion won't work with unit.to(unit, u.spectral_density). i know for sure that that equivalency will be needed for some translations, but i am not sure if translations between flux and surface brightness will ever occur, which would mean the custom defined equivalencies for those translations need to be supplied, and im not sure where u.spectral is needed. I am passing all of those to my new function just to be safe, because that's how the code originally was and i don't feel comfortable removing it yet until we verify they're unnecessary